pkg/validation/user.go

```diff
@@ -59,6 +59,16 @@ func ValidateUser(u *kubermaticv1.User) field.ErrorList {
 		}
 	}
 
+	// Ensure that `isAdmin` and `globalViewer` are not both true at the same time.
+	// These roles are mutually exclusive — a user can't be both an admin and a global viewer.
+	if isAdminAndGlobalViewer(u.Spec.IsAdmin, u.Spec.IsGlobalViewer) {
```

Comment:
```suggestion
	if isAdminAndGlobalViewer(u.Spec) {
```

Looks like you can simply pass the `UserSpec` in the function and let it decide internally which fields are important to do its job.